File: 김용성/Lv2/기능개발.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

typedef vector<int> vi;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vi v;
    for(int i = 0; i < speeds.size(); i++)
    {
        int remain = 100 - progresses[i];
        int days = remain / speeds[i];
        if(remain % speeds[i] != 0) days++;
        v.push_back(days);
    }
    for(int i = 0; i < v.size(); i++)
    {
        printf("%d ", v[i]);
    }
    printf("\n");
    for(int i = 0; i < speeds.size(); i++)
    {
        int start = v[i];
        cout <<"start : " << start;
        int cnt = 1;
        int j;
        for( j = i + 1; j < speeds.size(); j++ )
        {
            if(start >= v[j]) cnt++;
            else break;
        }
        answer.push_back(cnt);
        i = j - 1;
    }
    
    for(int i = 0; i < answer.size(); i++)
    {
        printf("%d ", answer[i]);
    }
    return answer;
}

File: 조선빈/Lv.1/기능개발.py
def solution(progresses, speeds):

    answer = []
    t = 0
    cnt = 0

    while len(progresses)> 0:
        if (progresses[0] + t*speeds[0]) >= 100: 
            progresses.pop(0)
            speeds.pop(0)
            cnt += 1

        else:
            if cnt > 0:
                answer.append(cnt)
                cnt = 0
            t += 1
    answer.append(cnt)
    return answer
